src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -183,19 +211,46 @@ public Task<ActionResult<IEnumerable<ProcessModel>>> GetProcesses()
 
                 return new OutputStreamResult(async (outputStream, token) =>
                 {
-                    await _diagnosticServices.StartLogs(outputStream, processInfo, duration, format, level, token);
+                    using var loggerFactory = new LoggerFactory();
+
+                    loggerFactory.AddProvider(new StreamingLoggerProvider(outputStream, format, level));
+
+                    var settings = new EventLogsPipelineSettings
+                    {
+                        Duration = duration,
+                        LogLevel = level,
+                        ProcessId = processInfo.Pid
+                    };
+                    await using EventLogsPipeline pipeline = new EventLogsPipeline(processInfo.Client, settings, loggerFactory);
+                    await pipeline.RunAsync(token);
+
                 }, contentType, downloadName);
             });
         }
 
-        private async Task<StreamWithCleanupResult> StartTrace(
+        private async Task<ActionResult> StartTrace(
             ProcessFilter? processFilter,
             MonitoringSourceConfiguration configuration,
             TimeSpan duration)
         {
             IProcessInfo processInfo = await _diagnosticServices.GetProcessAsync(processFilter, HttpContext.RequestAborted);
-            IStreamWithCleanup result = await _diagnosticServices.StartTrace(processInfo, configuration, duration, this.HttpContext.RequestAborted);
-            return new StreamWithCleanupResult(result, "application/octet-stream", FormattableString.Invariant($"{GetFileNameTimeStampUtcNow()}_{processInfo.Pid}.nettrace"));
+
+            return new OutputStreamResult(async (outputStream, token) =>
+            {
+                Func<Stream, CancellationToken, Task> streamAvailable = async (Stream eventStream, CancellationToken token) =>
+                {
+                    await eventStream.CopyToAsync(outputStream, 0x1000, token);
```

Comment:
I will add an option to the configuration.

Reply:
After looking at this some more, I'm going to match the buffer size to [FileStreamResult](https://github.com/iamoldli/AspNetCoreSource/blob/d2a6d4e25639031c5d02738c0e9c6f03d63c0bd3/src/AspNetCore/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs#L22 ) since this is the intended behavior.